proof/invariant-abstract/ARM/ArchDetSchedDomainTime_AI.thy

```diff
@@ -56,25 +56,39 @@ crunch domain_time_inv [wp, DetSchedDomainTime_AI_assms]: arch_perform_invocatio
 crunch domain_list_inv [wp, DetSchedDomainTime_AI_assms]: arch_perform_invocation "\<lambda>s. P (domain_list s)"
   (wp: crunch_wps check_cap_inv)
 
+lemma timer_tick_valid_domain_time:
+  "\<lbrace> \<lambda>s :: det_ext state. 0 < domain_time s \<rbrace>
+   timer_tick
+   \<lbrace>\<lambda>x s. domain_time s = 0 \<longrightarrow> scheduler_action s = choose_new_thread\<rbrace>" (is "\<lbrace> ?dtnot0 \<rbrace> _ \<lbrace> _ \<rbrace>")
+  unfolding timer_tick_def
+  supply if_split[split del]
+  supply ethread_get_wp[wp del]
+  supply if_apply_def2[simp]
+  apply (wpsimp
+           wp: reschedule_required_valid_domain_time hoare_vcg_const_imp_lift gts_wp
+               (* unless we hit dec_domain_time we know ?dtnot0 holds on the state, so clean up the
+                  postcondition once we hit thread_set_time_slice *)
+               hoare_post_imp[where Q="\<lambda>_. ?dtnot0" and R="\<lambda>_ s. domain_time s = 0 \<longrightarrow> X s"
+                                and a="thread_set_time_slice t ts" for X t ts]
+               hoare_drop_imp[where f="ethread_get t f" for t f])
```

Comment:
This might be a whitespace change, reindent or moving things around, don't think I did anything this round on it, though I did possibly write the original.